### PR TITLE
Cu 1z4mgmq separate web framework from api

### PR DESCRIPTION
**Pull request by rkamysz**
This PR includes:

- separation of the API logic from the web framework
- new unit tests
- new API tests
- small bug fixes

### CODE REVIEW

File: src/api-handlers/asset/domain/use-cases/get-assets.use-case.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      return this.assetRepository.getManyByAssetId(assetIds);
+    }
+
+    if (owner) {
+      return this.assetRepository.getAssets(
+        GetAssetsQueryModel.create(owner, schema, offset, limit)
+      );
+    }
+
+    // TODO question should we log missing options or return failure
```
Comment (ahmadtirmazi) [inline src/api-handlers/asset/domain/use-cases/get-assets.use-case.ts:41]: May be we can remove this

File: src/api-handlers/asset/routes/get-assets.route.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import { GetAssetsInput } from '../domain/models/get-assets.input';
+import { GetAssetsOutput } from '../domain/models/get-assets.output';
+import { AssetsNotFoundError } from '@common/assets/domain/errors/assets-not-found.error';
+
+/**
+ *
+ * @param {Result<Asset[]>} result
+ * @returns
+ */
+export const parseHandlerResultToResponse = (result: Result<Asset[]>) => {
```
Comment (ahmadtirmazi) [inline src/api-handlers/asset/routes/get-assets.route.ts:14]: Probably we can rename this from Handler to `controller`

File: src/api-handlers/api.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    NftsController.Token
+  );
+  const tokenController: TokenController = apiIoc.get<TokenController>(
+    TokenController.Token
+  );
+
+  // Mount routes
+  FastifyRoute.mount(
+    api,
+    GetAssetRoute.create(assetController.getAssets.bind(assetController))
```
[the remark this answers is not in the document]
Comment (rkamysz) [reply src/api-handlers/api.ts]: done

File: src/api-handlers/asset/domain/use-cases/get-assets.use-case.ts (hunk 1 above)
Comment (rkamysz) [reply src/api-handlers/asset/domain/use-cases/get-assets.use-case.ts]: removed

File: src/api-handlers/asset/routes/get-assets.route.ts (hunk 2 above)
Comment (rkamysz) [reply src/api-handlers/asset/routes/get-assets.route.ts]: done

**Pull request merged**